src/biotite/structure/rings.py

```diff
@@ -268,6 +273,122 @@ def find_stacking_interactions(
     ]
 
 
+def find_pi_cation_interactions(
+    atoms,
+    distance_cutoff=5.0,
+    angle_tol=np.deg2rad(30.0),
+):
+    """
+    Find pi-cation interactions between aromatic rings and cations.
+
+    Parameters
+    ----------
+    atoms : AtomArray
+        The atoms to be searched for pi-cation interactions.
+        Requires an associated :class:`BondList` and ``charge`` annotation.
+    distance_cutoff : float, optional
+        The cutoff distance between ring centroid and cation.
+    angle_tol : float, optional
+        The tolerance for the angle between the ring plane normal
+        and the centroid-cation vector. Perfect pi-cation interaction
+        has 0° angle (perpendicular to ring plane).
+        Given in radians.
+
+    Returns
+    -------
+    interactions : list of tuple(ndarray, int)
+        The pi-cation interactions between aromatic rings and cations.
+        Each element in the list represents one pi-cation interaction.
+        The first element of each tuple represents atom indices of the
+        aromatic ring, the second element is the atom index of the cation.
+
+    See Also
+    --------
+    find_aromatic_rings : Used for finding the aromatic rings in this function.
+    find_stacking_interactions : Find pi-stacking interactions between rings.
+
+    Notes
+    -----
+    This function requires the ``charge`` annotation to identify cations.
+    The charge annotation can be added when loading structures with
+    ``extra_fields=["charge"]``.
+
+    The conditions for pi-cation interactions are:
+
+        - The distance between ring centroid and cation must be within
+          `distance_cutoff`. Open Drug Discovery Toolkit (ODDT) uses 5.0 Å,
+          whereas Protein-Ligand Interaction Fingerprints (ProLIF) uses 4.5 Å.
+        - The angle between the ring plane normal and the centroid-cation
+          vector must be within `angle_tol` of 0° (perpendicular to plane).
+
+    Examples
+    --------
+    >>> structure = load_structure("protein_ligand.cif", include_bonds=True, extra_fields=["charge"])
```

Comment:
In the Biotite test suite the doctests are actually executed, so this line will fail the test, as `protein_ligand.cif` does not exist. However you can use the predefined `path_to_structures` variable that actually points to `tests/structure/data` to select an existing structure file. You can take an inspiration from this docstring:

https://github.com/biotite-dev/biotite/blob/5ac8154335f5de8b7952a9c4b55345d2cc1fa416/src/biotite/structure/rings.py#L172-L187